### PR TITLE
UML-1094 Viewer add organisation 

### PR DESCRIPTION
**Pull request by GemTay**
# Purpose

Organisations must now enter their organisation name on the check code page, to allow actors to see which organisations have used a code to view their LPA (actor side is a separate ticket)

Fixes UML-1094

## Approach

Created a new organisation form and added it to the check code page. I changed the lpa service functions to take in a new parameter of the organisation that defaults to null if not provided. I removed the summary/full parameter and added logic to change the route (full or summary) based off whether the organisation was provided. 

## Checklist

* [x] I have performed a self-review of my own code
* [ ] I have added relevant logging with appropriate levels to my code
* [ ] I have updated documentation (Confluence/GitHub wiki/tech debt doc) where relevant
* [x] I have added tests to prove my work
* [ ] I have added welsh translation tags and updated translation files
* [ ] The product team have tested these changes

### CODE REVIEW

File: service-api/app/src/App/src/Handler/ViewerCodeFullHandler.php [hunk 1]
```diff
@@ -38,11 +38,11 @@ public function handle(ServerRequestInterface $request): ResponseInterface
     {
         $params = $request->getParsedBody();
 
-        if (!isset($params['code']) || !isset($params['name'])) {
+        if (!isset($params['code']) || !isset($params['name']) || !isset($params['organisation'])) {
             throw new RuntimeException("'code' and 'name' are required fields.");
```
Comment (cooperaj) [inline service-api/app/src/App/src/Handler/ViewerCodeFullHandler.php:42]:
```suggestion
            throw new RuntimeException("'code', 'name'  and 'organisation' are required fields.");
```

File: service-api/app/src/App/src/Service/Lpa/LpaService.php [hunk 2]
```diff
@@ -166,11 +166,10 @@ public function getAllForUser(string $userId): array
      *
      * @param string $viewerCode
      * @param string $donorSurname
-     * @param bool $logActivity
+     * @param string|null $organisation
      * @return array|null
-     * @throws \Exception
      */
-    public function getByViewerCode(string $viewerCode, string $donorSurname, bool $logActivity): ?array
+    public function getByViewerCode(string $viewerCode, string $donorSurname, string $organisation = null): ?array
```
Comment (cooperaj) [inline service-api/app/src/App/src/Service/Lpa/LpaService.php:172]:
```suggestion
    public function getByViewerCode(string $viewerCode, string $donorSurname, ?string $organisation = null): ?array
```

File: service-api/docker/seeding/start.sh [hunk 3]
```diff
@@ -57,13 +57,14 @@ if ! [[ -z "${AWS_ENDPOINT_DYNAMODB}" ]]; then
     --global-secondary-indexes IndexName=SiriusUidIndex,KeySchema=["{AttributeName=SiriusUid,KeyType=HASH},{AttributeName=Expires,KeyType=RANGE}"],Projection="{ProjectionType=ALL}",ProvisionedThroughput="{ReadCapacityUnits=10,WriteCapacityUnits=10}"
 
     aws dynamodb create-table \
-    --attribute-definitions AttributeName=ViewerCode,AttributeType=S AttributeName=Viewed,AttributeType=S \
+    --attribute-definitions AttributeName=ViewerCode,AttributeType=S AttributeName=Viewed,AttributeType=S\
```
Comment (cooperaj) [inline service-api/docker/seeding/start.sh:60]:
```suggestion
    --attribute-definitions AttributeName=ViewerCode,AttributeType=S AttributeName=Viewed,AttributeType=S \
```

File: service-front/app/src/Common/src/Service/Lpa/LpaService.php [hunk 4]
```diff
@@ -107,21 +104,30 @@ public function getLpaById(string $userToken, string $actorLpaToken): ?ArrayObje
...
      * @param string $shareCode
      * @param string $donorSurname
-     * @param bool $response
+     * @param string|null $organisation
      * @return ArrayObject|null
-     * @throws ApiException|Exception
+     * @throws Exception
      */
-    public function getLpaByCode(string $shareCode, string $donorSurname, bool $response = self::SUMMARY): ?ArrayObject
+    public function getLpaByCode(string $shareCode, string $donorSurname, string $organisation = null): ?ArrayObject
```
Comment (cooperaj) [inline service-front/app/src/Common/src/Service/Lpa/LpaService.php:111]:
```suggestion
    public function getLpaByCode(string $shareCode, string $donorSurname, ?string $organisation = null): ?ArrayObject
```

File: service-front/app/src/Viewer/src/Form/Organisation.php [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+use Laminas\InputFilter\InputFilterProviderInterface;
+use Laminas\Validator\NotEmpty;
+use Mezzio\Csrf\CsrfGuardInterface;
+
+class Organisation extends AbstractForm implements InputFilterProviderInterface
+{
+    public const FORM_NAME = 'organisation_name';
+
+    protected array $messageTemplates = [
+        self::NOT_SAME => "Do you want to continue?" .
```
Comment (cooperaj) [inline service-front/app/src/Viewer/src/Form/Organisation.php:18]: Whats this message?

**Review by cooperaj — CHANGES_REQUESTED** [review]
Few tweaks

File: service-front/app/src/Viewer/src/Form/Organisation.php (hunk 5 above)
Comment (GemTay) [reply service-front/app/src/Viewer/src/Form/Organisation.php:18]: @cooperaj I believe its the error message for the CSRF validation. Its the same on the ShareCode.php form

**Review by SeemaMenon — APPROVED** [review]
lgtm

**Review by cooperaj — APPROVED** [review]

**Pull request merged**